### PR TITLE
feat!: add node selectors features

### PR DESCRIPTION
**Pull request by barnabasbusa**

### CODE REVIEW

**Comment by barnabasbusa** [comment]
Requires kt version  0.86.17

File: .github/tests/node-selectors.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    cl_client_type: teku
+    cl_split_mode_enabled: true
+    node_selectors: {
+      "kubernetes.io/hostname": testing-1,
+    }
+  - el_client_type: reth
+    cl_client_type: teku
+    cl_split_mode_enabled: true
+global_node_selectors: {
+  " kubernetes.io/hostname": testing-2,
```
Comment (h4ck3rk3y) [inline .github/tests/node-selectors.yaml:12]: extra space

Comment (h4ck3rk3y) [reply .github/tests/node-selectors.yaml:12]: seems like this test isn't running? otherwise why isn't it failing

Comment (h4ck3rk3y) [reply .github/tests/node-selectors.yaml:12]: this should be moved into Circle and run as a k8s test!

**Review by h4ck3rk3y — APPROVED** [review]

File: .github/tests/node-selectors.yaml (hunk 1 above)
Comment (h4ck3rk3y) [inline .github/tests/node-selectors.yaml:12]:
```suggestion
  "kubernetes.io/hostname": testing-2,
```

Comment (barnabasbusa) [reply .github/tests/node-selectors.yaml]: If we move it to Circleci, then its going to continuously fail. I rather leave it as is, and let it not fail.

**Pull request merged**